src/decoder.rs

```diff
@@ -179,22 +124,49 @@ fn decode_c_reg_based(raw: usize) -> Instr {
         0b111 => {
             let rs2 = rd_rs2;
             let imm = 0b000 | (raw >> 7) & 0b111000 | ((raw << 1) & 0b11000000);
-            Instr::CSd { rs1, rs2, imm }
+
```

Comment:
Small style nit: you don't have a while line here in the other cases, let's keep things consistent :)

```suggestion
```